source/develop/release-management/features/external-network-provider.html.md

```diff
@@ -0,0 +1,402 @@
+* Add subnet (POST subnets)
+
+External network provider interactions when removing a subnet (Networks -> Subnets tab -> Remove):
+* Add subnet (DELETE subnets/<id>)
+
+Opening the subnets subtab causes a request for the list of subnets
+* Get subnets (GET subnets)
+
+#### Setup networks dialog (UI) - optional
+The setup network dialog could be used to modify external networks. Some of the possible operations might include:
+
+* binding external network to a host NIC
+
+* binding external network to a bond, and modifying the bond properties
+
+* modyfing external network properties
+
+This option can however be quite problematic, as it would limit the allowed implementations of how an external network is provided by an external party. Modifications of networks set up by the external provider could also lead to unexpected problems.
+
+
+### VM NIC lifecycle
+
+The most important function of this feature is to handle the lifecycle of VM NICs connected to a network defined outside of oVirt.
+
+
+#### VM NIC provisioning
+
+When a new nic is added to a VM the following actions are being executed:
+
+Port allocation:
+
+* the engine checks if the port for the VM NIC exists (GET ports)
+* if no port exists, the engine will create a new port (POST ports)
+* if the port already exists, the engine will update it with information about new allocation (POST ports)
+* in both cases the new/updated port is identified by a <PORT ID> returned by the external network provider
+* the engine will send a nic plug request to VDSM, passing the <PORT ID> as one of the parameters.
+* on VDSM, the VIF Driver, invoked using the VDSM before_nic_hotplug hook, will connect the VM NIC to the network provided by the external provider
+
+![](external_network_provider_schema1.jpg "fig:external_network_provider_schema1.jpg")
+
+#### VM NIC unplug
+
+When the nic is unplugged from the VM, the following actions are being executed:
+
+* on VDSM, the VIF Driver, invoked using the VDSM before_nic_hotunplug hook, will unplug the VM NIC from the network provided by the external provider. The nic is identified by the same <PORT ID> a during nic plug
+
+#### VM NIC delete
+
+When a nic is already unplugged, it can be deleted from a VM. This will trigger the following actions:
+
+* the engine will try to locate the port on the external provider (GET ports)
+* the engine will delete the port from the external provider (DELTE ports)
+
+#### Migrate a VM to another host
+
+When a VM is migrated to another host the following actions are executed:
+
+* the VIF driver, invoked by the VDSM before_migration_source disconnects the NIC from the external network on the source VDSM
+* the VIF driver, invoked by the VDSM before_migration_destination connects the NIC ti\o the external network on the destination VDSM
+
+#### VM NIC provisioning synchronization issues
+
+Care must be taken for the VIF driver to leave the networking operations in a finished state before exiting. 
```

Comment:
add a reference to https://blueprints.launchpad.net/neutron/+spec/nova-event-callback ?